lib/rubocop/cop/rspec/file_fixture.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Favor the use of `file_fixture`.
+      #
+      # `RSpec` provides the convenience method
+      #
+      # [source,ruby]
+      # ----
+      # file_fixture('path/some_file.csv')
+      # ----
+      #
+      # which (by default) is equivalent to
+      #
+      # [source,ruby]
+      # ----
+      # Rails.root.join('spec/fixtures/files/path/some_file.csv')
+      # ----
+      #
+      # @example
+      #   # bad
+      #   "#{Rails.root}/spec/fixtures/path/some_file.csv"
+      #   Rails.root.join('spec/fixtures/path/some_file.csv')
+      #   Rails.root.join('spec', 'fixtures', 'path', 'some_file.csv')
+      #
+      #   "#{Rails.root}/spec/fixtures/files/some_file.csv"
+      #   Rails.root.join('spec/fixtures/files/some_file.csv')
+      #   Rails.root.join('spec', 'fixtures', 'files', 'some_file.csv')
+      #
+      #   # good
+      #   file_fixture('../path/some_file.csv').path
+      #   file_fixture('../path/some_file.csv')
+      #   file_fixture('../path/some_file.csv')
+      #
+      #   file_fixture('some_file.csv').path
+      #   file_fixture('some_file.csv')
+      #   file_fixture('some_file.csv')
+      #
+      class FileFixture < Base
+        extend AutoCorrector
+
+        MSG = 'Use `file_fixture`.'
+
+        RESTRICT_ON_SEND = %i[join root].freeze
+
+        DEFAULT_FILE_FIXTURE_PATTERN = %r{^spec/fixtures/files/}.freeze
+        DEFAULT_FIXTURE_PATTERN = %r{^spec/fixtures/}.freeze
+
+        # @!method file_io?(node)
+        def_node_matcher :file_io?, <<~PATTERN
+          (send
+            (const {nil? cbase} :File) {:binread :binwrite :open :read :write} $...)
+        PATTERN
+
+        # @!method rails_root_join(node)
+        def_node_matcher :rails_root_join, <<~PATTERN
+          (send
+            (send
+              (const {nil? cbase} :Rails) :root) :join (str $_)+)
```

Comment:
What if it's not a literal string passed to `join`? E.g. a string variable or an interpolated string.